### PR TITLE
feat: Shifted CI from Travis to Github Actions

### PR DESCRIPTION
**Pull request by Jawayria**
This is a part of Django 3.2 upgrade process as mentioned in this doc [Django 3.2 Upgrade: Steps to Upgrade a Repository](https://openedx.atlassian.net/wiki/spaces/AC/pages/1704591765/Django%2B3.2%2BUpgrade%2BSteps%2Bto%2BUpgrade%2Ba%2BRepository)

JIRA: https://openedx.atlassian.net/browse/BOM-2724

### CODE REVIEW

**Review by UsamaSadiq — APPROVED** [review]

**Comment by Jawayria** [comment]
@edx/enterprise-markhors please review this PR and let me know if any changes are required

File: .github/workflows/pypi-publish.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - name: setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Install pip
+        run: pip install -r requirements/pip.txt
+
+      - name: Install Dependencies
+        run: pip install setuptools wheel
```
Comment (mraarif) [inline .github/workflows/pypi-publish.yml:24]: we don't have to install these packages explicitly, these should be listed in `pip.txt`

File: .github/workflows/pypi-publish.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+      - name: Build package
+        run: python setup.py sdist bdist_wheel
+
+      - name: Publish to PyPi
+        uses: pypa/gh-action-pypi-publish@master
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_UPLOAD_TOKEN }}
```
Comment (mraarif) [inline .github/workflows/pypi-publish.yml:33]: file should end with a newline

File: requirements/pip.in [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Core dependencies for installing other packages
+
+pip
+setuptools
+wheel
```
Comment (mraarif) [inline requirements/pip.in:5]: same here

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      env:
+        TOXENV: ${{ matrix.toxenv }}
+      run: tox
+
+    - name: Run Coverage
+      if: matrix.python-version == '3.8' && matrix.toxenv=='django22'
+      uses: codecov/codecov-action@v1
+      with:
+        flags: unittests
+        fail_ci_if_error: true
```
Comment (mraarif) [inline .github/workflows/ci.yml:43]: file should end with a newline

**Review by mraarif — APPROVED** [review]
please squash the commits before you merge

**Comment by Jawayria** [comment]
> please squash the commits before you merge

Okay

File: taxonomy/algolia/client.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
```
Comment (saleem-latif) [inline taxonomy/algolia/client.py:1]: These files were added in https://github.com/edx/taxonomy-connector/pull/67/files#diff-16d854d1f34428bdb104335e5714e693bb434a903fbcbbc5169186d418f70dbdR1 why are they appearing in this PR as well?
maybe you need to rebase?

**Review by saleem-latif — COMMENTED** [review]
A question about the duplicate changes that are already present in taxonomy-connector, once that is addressed 👍🏼

**Comment by Jawayria** [comment]
@saleem-latif Please merge the PR. I don't have merge access on this repo

**Pull request merged**